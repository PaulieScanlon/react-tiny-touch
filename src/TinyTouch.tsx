/* tslint:disable */
import * as React from 'react'
import { useState } from 'react'
/* tslint:enable */

export type TTouch = () => React.ReactElement

export const TinyTouch: React.SFC = ({ children }) => {
  const [touch, setTouch] = useState(false)

  if ('ontouchstart' in window && !touch) {
    setTouch(true)
  }

  if (typeof children === 'function') {
    return children(touch)
  }

  return touch ? children : null
}
