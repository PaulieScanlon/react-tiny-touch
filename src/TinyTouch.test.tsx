import * as React from 'react'

import { shallow, mount } from 'enzyme'
import { TinyTouch } from './TinyTouch'

describe('TinyTouch', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <TinyTouch>
        <div />
      </TinyTouch>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders children if hasTouch is detected', () => {
    const window = {}
    Object.defineProperty(window, 'ontouchstart', {
      value: true
    })
    const wrapper = mount(
      <TinyTouch>
        <div />
      </TinyTouch>
    )
    expect(wrapper.find('div'))
  })

  it('renders the correct element(s) if hasTouch is detected', () => {
    const window = {}
    Object.defineProperty(window, 'ontouchstart', {
      value: true
    })

    const wrapper = mount(
      <TinyTouch>{hasTouch => (hasTouch ? <div /> : <span />)}</TinyTouch>
    )
    expect(wrapper.find('div'))
  })

  it('renders the correct element(s) if hasTouch is not detected', () => {
    const window = {}
    Object.defineProperty(window, 'ontouchstart', {
      value: false
    })

    const wrapper = mount(
      <TinyTouch>{hasTouch => (hasTouch ? <div /> : <span />)}</TinyTouch>
    )
    expect(wrapper.find('span'))
  })
})
