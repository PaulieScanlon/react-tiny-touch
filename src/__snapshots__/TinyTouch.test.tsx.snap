// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TinyTouch renders correctly 1`] = `""`;
